### PR TITLE
update report permissions and update action versions

### LINKED ISSUE #1488: [🐛 BUG] Coverage report is failing
### What went wrong? 🤔

The cover report does not have the correct authorization to update the PR and failed.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [X] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by florian-vuillemot**
close #1488 

- update action versions
- add permission for coverage
- re enable packaging on window

### CODE REVIEW

File: .github/workflows/overall-tests.yml [hunk 1]
```diff
@@ -7,6 +7,9 @@ on:
     branches: [ develop, dev/*, release/* ]
   workflow_dispatch:
 
+permissions:
+  pull-requests: write
+
```
Comment (jrobinAV) [inline .github/workflows/overall-tests.yml:12]: I don't think that was the issue. Was it?
I believe the permission was granted directly at the GitHub organization level. I am not fully certain, though.

**Review by jrobinAV — DISMISSED** [review]

File: .github/workflows/overall-tests.yml (hunk 1 above)
Comment (florian-vuillemot) [reply .github/workflows/overall-tests.yml:12]: That was the issue on taipy-config. This is due to the GITHUB_TOKEN [accesses](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token).

Comment (jrobinAV) [reply .github/workflows/overall-tests.yml:12]: All right. Thx.

**Review by dinhlongviolin1 — DISMISSED** [review]
LGTM

**Review by trgiangdo — DISMISSED** [review]

**Review by dinhlongviolin1 — DISMISSED** [review]
LGTM! Please merge it so everyone can enjoy it

**Pull request merged**